unittest/.clang-tidy

```diff
@@ -0,0 +1,27 @@
+---
+Checks:          '-*,readability-function-size'
+WarningsAsErrors: '*'
+HeaderFilterRegex: 'unittest/.*'
+CheckOptions:    
+  - key:             readability-braces-around-statements.ShortStatementLines
+    value:           2
+
+  # extremly high limits, they are just here to ensure the situation doesn't
+  # get worse than it already is
+  # If you hit the limits, please change the code and not the limits!!
+  # Note: some limits disabled due to TEST_SUITE macro.
+  # It generates hundreds of statements, branches and variables.
+  - key:             readability-function-size.LineThreshold
+    value:           130
```

Comment:
FWIW, I don't consider 130 to be "extremely high". Which limit would you like to have?

Reply:
You are correct, only the 99999 a few lines later is "extremely high". I'll make this more clear.

Reply:
I just searched **a little** what different people advice on function length. Some respected folk say 10 lines. Others say it doesn't matter at all (big monitors etc).
Important part is that a function does only one thing - but that cannot be measured by tools.

A word on limits:
I recently had to deal with such a limit and spend 2 days refactoring a working pull request. It was a 80 char line limit and my function took up 60 of those. Because it did two things and I at least didn't want to hide that from the function name.
It was indeed way better afterwards! But not every contributor has those 2 days. And especially if you don't enjoy clean code as I do this is most definitely extremely demotivating. So I would keep the limit way above 10 here ;-)

Reply:
> especially if you don't enjoy clean code as I do

You make it sound like there are people who don't enjoy clean code, but I very much doubt that many people dislike clean code. But what's clean or not is subjective, so your clean code is not necessarily my clean code.

Reply:
Absolutely true. What I meant is that even then it differs how much people are willing to invest to arrive there.

Back to the topic. I guess 130 is fine for now?! When you hit 130 lines it's time to split your function. It doesn't prevent anyone from doing so before 130 lines :-)
The difficulty is that it's not possible to set exceptions. So the limit should probably remain at 130 and not be lowered.

Reply:
> I guess 130 is fine for now?!

It's fine for now.

>  When you hit 130 lines it's time to split your function.

Again, that's inherently subjective. I'm personally much more tolerant to long functions than complex functions, even though there of course often is correlation between long and complex.